services/web/client/source/class/qxapp/auth/BaseAuthPage.js

```diff
@@ -0,0 +1,101 @@
+/*
+
+  TODO: add alternative logins
+*/
+
+/**
+ * This is a basic Auth-Page with common functionality
+ *
+ *  - Fixed-size widget with header (title/logo), body and footer (buttons)
+ *  - Positioned at the upper center of the root document's window
+ */
+qx.Class.define("qxapp.auth.BaseAuthPage", {
+  extend: qx.ui.container.Composite,
+  include: [
+    qxapp.auth.MAuth
+  ],
+  type: "abstract",
+
+  statics: {
+
+    addPageToRoot: function(page) {
+      // TODO: should be automatically reposition of document size changed!?
+
+      let top = parseInt((qx.bom.Document.getHeight() - page.getHeight()) / 4, 10);
+      let left = parseInt((qx.bom.Document.getWidth() - page.getWidth()) / 2, 10);
+
+      let app = qx.core.Init.getApplication();
+      app.getRoot().removeAll();
+      app.getRoot().add(page, {
+        top: top,
+        left: left
+      });
+    }
+  },
+
+
+  construct: function() {
+    this.base(arguments);
+
+    // Setup children's layout and the auth page widget dimensions
+    const spacing = 10;
+    this.setLayout(new qx.ui.layout.VBox(spacing));
+    this.set({
+      width: 300,
+      height: 250
+    });
+
+    // layout values
+    this._widthBtn = parseInt((this.getWidth() - 30) / 2, 10);
+    this._marginFooter = this._marginHeader = 3*spacing;
+
+    this._buildPage();
```

Comment:
@oetiker Please comment on this mechanism to "pass" pages ... I am still not convinced this is the best way to go. All input is welcome.